### ISSUE TITLE
Unable to find extruder definition with the id [flsun_v400_extruder_0]

### ISSUE THREAD

**Issue by lloydpick** [phnxdrgn536/FLSun-v400#10]
Clean install, ran the bat file. Crashes Cura when I go to add the printer.

```2023-01-09 18:47:53,658 - INFO - [MainThread] cura.Settings.MachineManager.addMachine [425]: Trying to add a machine with the definition id [flsun_v400]
2023-01-09 18:47:53,826 - WARNING - [MainThread] LocalContainerProvider.LocalContainerProvider._saveCachedDefinition [308]: The definition cache for definition flsun_v400 failed to pickle.
2023-01-09 18:47:54,138 - DEBUG - [MainThread] cura.Machines.ContainerTree.__getitem__ [120]: Adding container tree for flsun_v400 took 0.30750083923339844 seconds.
2023-01-09 18:47:54,193 - DEBUG - [MainThread] UM.Settings.DefinitionContainer.getConfigurationTypeFromSerialized [265]: Could not get configuration type: Extra data: line 1 column 4 (char 3)
2023-01-09 18:47:54,194 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]: Exception: Unable to deserialize metadata for container C:\Users\lloyd\AppData\Roaming\cura\5.2\extruders\flsun_v400_extruder_0.def.json: flsun_v400_extruder_0: Extra data: line 1 column 4 (char 3)
2023-01-09 18:47:54,195 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]: Traceback (most recent call last):
2023-01-09 18:47:54,195 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "C:\Program Files\Ultimaker Cura 5.2.1\share\uranium\plugins\LocalContainerProvider\LocalContainerProvider.py", line 159, in loadMetadata
2023-01-09 18:47:54,196 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:     result_metadatas = clazz.deserializeMetadata(f.read(), container_id) #pylint: disable=no-member
2023-01-09 18:47:54,196 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "UM\Settings\DefinitionContainer.py", line 360, in deserializeMetadata
2023-01-09 18:47:54,196 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:     serialized = cls._updateSerialized(serialized) #Update to most recent version.
2023-01-09 18:47:54,197 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "UM\Settings\Interfaces.py", line 168, in _updateSerialized
2023-01-09 18:47:54,197 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:     version = cls.getVersionFromSerialized(serialized)
2023-01-09 18:47:54,198 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "UM\Settings\DefinitionContainer.py", line 284, in getVersionFromSerialized
2023-01-09 18:47:54,198 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:     parsed = json.loads(serialized, object_pairs_hook = collections.OrderedDict)
2023-01-09 18:47:54,199 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "json\__init__.py", line 359, in loads
2023-01-09 18:47:54,199 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]:   File "json\decoder.py", line 340, in decode
2023-01-09 18:47:54,200 - ERROR - [MainThread] LocalContainerProvider.LocalContainerProvider.loadMetadata [165]: json.decoder.JSONDecodeError: Extra data: line 1 column 4 (char 3)
2023-01-09 18:47:54,200 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createExtruderStackWithDefaultSetup [108]: Exception: Unable to find extruder definition with the id [flsun_v400_extruder_0]
2023-01-09 18:47:54,201 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createExtruderStackWithDefaultSetup [108]: Traceback (most recent call last):
2023-01-09 18:47:54,201 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createExtruderStackWithDefaultSetup [108]:   File "cura\Settings\CuraStackBuilder.py", line 104, in createExtruderStackWithDefaultSetup
2023-01-09 18:47:54,202 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createExtruderStackWithDefaultSetup [108]:     extruder_definition = registry.findDefinitionContainers(id = extruder_definition_id)[0]
2023-01-09 18:47:54,202 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createExtruderStackWithDefaultSetup [108]: IndexError: list index out of range
2023-01-09 18:47:54,202 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: Exception: Failed to create an extruder stack for position 0: Unable to find extruder definition with the id [flsun_v400_extruder_0]
2023-01-09 18:47:54,203 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: Traceback (most recent call last):
2023-01-09 18:47:54,203 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:   File "cura\Settings\CuraStackBuilder.py", line 104, in createExtruderStackWithDefaultSetup
2023-01-09 18:47:54,204 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:     extruder_definition = registry.findDefinitionContainers(id = extruder_definition_id)[0]
2023-01-09 18:47:54,204 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: IndexError: list index out of range
2023-01-09 18:47:54,204 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: 
2023-01-09 18:47:54,205 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: During handling of the above exception, another exception occurred:
2023-01-09 18:47:54,205 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: 
2023-01-09 18:47:54,205 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: Traceback (most recent call last):
2023-01-09 18:47:54,206 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:   File "cura\Settings\CuraStackBuilder.py", line 67, in createMachine
2023-01-09 18:47:54,206 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:     cls.createExtruderStackWithDefaultSetup(new_global_stack, position)
2023-01-09 18:47:54,206 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:   File "cura\Settings\CuraStackBuilder.py", line 109, in createExtruderStackWithDefaultSetup
2023-01-09 18:47:54,207 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]:     raise IndexError(msg)
2023-01-09 18:47:54,207 - ERROR - [MainThread] cura.Settings.CuraStackBuilder.createMachine [69]: IndexError: Unable to find extruder definition with the id [flsun_v400_extruder_0]
2023-01-09 18:47:54,207 - WARNING - [MainThread] cura.Settings.MachineManager.addMachine [438]: Failed creating a new machine!```

**Comment by lloydpick** [comment]
Ok, found the issue, since restructuring the URLs are incorrect. Also the bat is downloading ABS profiles for PETG. Both fixed in #11

**Comment by phnxdrgn536** [comment]
fixed in #11
